src/mlpack/tests/loss_functions_test.cpp

```diff
@@ -578,4 +579,43 @@ BOOST_AUTO_TEST_CASE(HingeEmbeddingLossTest)
   BOOST_REQUIRE_EQUAL(output.n_rows, input.n_rows);
   BOOST_REQUIRE_EQUAL(output.n_cols, input.n_cols);
 }
+
+/**
+ * Simple test for the l1 loss function.
+ */
+BOOST_AUTO_TEST_CASE(SimpleL1LossTest)
+{
+  arma::mat input1, input2, output, target1, target2;
+  L1Loss<> module(false);
+
+  // Test the Forward function on a user generator input and compare it against
+  // the manually calculated result.
+  input1 = arma::mat("0.5 0.5 0.5 0.5 0.5 0.5 0.5");
+  target1 = arma::zeros(1, 7);
+  double error1 = module.Forward(input1, target1);
+  BOOST_REQUIRE_EQUAL(error1, 3.5);
+
+  input2 = arma::mat("0 1 1 0 1 0 0 1");
+  target2 = arma::mat("0 1 1 0 1 0 0 1");
+  double error2 = module.Forward(input2, target2);
+  BOOST_REQUIRE_CLOSE(error2, 0.0, 0.00001);
+
+  // Test the Backward function.
+  module.Backward(input1, target1, output);
+  for (double el : output)
+  {
```

Comment:
We don't need the braces here or in the other for loop

Reply:
Done